### DISCUSSION TITLE
Data exchange Venta LW74

### DISCUSSION THREAD

**Discussion by wimweiss** [Michsior14/ha-venta#16] [General]
dear reader,

my Venta LW74 is integrated in HA. Some data is displayed correctly compared to the data in Venta app. The time related data is not displayed correctly:
- service time in HA 702 min and in Venta app 10 days
- disk replacement in HA 913 min and in Venta app 115 days
- cleaning in HA 913 min and in Venta app 176 days

Also level UOM is missing and probably not correct. When empty displayed level is 46.566

Pls help me out.

**Comment by Michsior14** [comment]
Hey,
All the data are returned directly from the device without any modifications. They are not the same as in app simply because I don't know what kind of magic Venta uses to calculate them, especially regarding the days you mentioned. In my initial observations I assumed those are minutes, but in reality I have no idea. If you have any details then please provide them so we can improve this.

There is no unit for water level for exactly the same reason, the value is always super high (probably voltage).

**Comment by wimweiss** [comment]
Hi, thx for reaction.
I also noticed the timers are increasing, while expecting decreasing numbers.
I'll send a mail to Venta.

**Reply by Offi83** [reply]
Hi, if you look how the value is increasing, you would see that it adds +1 every 10min.
I use three template sensors to show the remaining days to Service, Disc change and Cleaning:

venta_lw74_timetodisc (every four month, max seen value 17426)
`{{ 121 - (((states('sensor.venta_lw73_lw74_disc_ionization_time') | float(0)) / 6 / 24) | round(0)) }}`

venta_lw74_timetoclean (every six month, max seen value 26210)
`{{ 182 - (((states('sensor.venta_lw73_lw74_cleaning_time') | float(0)) / 6 / 24) | round(0)) }}`

venta_lw74_timetoservice (every 14 days, max seen value 26210)
`{{ 14 - (((states('sensor.venta_lw73_lw74_service_time') | float(0)) / 6 / 24) | round(0)) }}`

Hope this helps. It would be very nice to get these values and "binary_sensor.venta_lw73_lw74_needs_newdisc" directly from the integration.

**Reply by Michsior14** [reply]
@Offi83 Added as a part of the latest release.

**Comment by rgr101** [comment]
I am also wondering about some other entities, e.g. 

- select.venta_led_strip_mode (intern  exterNoWater  internNoWater  extern) 
Does anyone know what the options mean/do?

- sensor.venta_water_level (xx.xxxx)
What does this number mean? I haven't found a sensor that can measure a continuous water level, only an open contact. So I assume that it is de facto a digital value; and this entity could represent a measuring current in uA or mA (in my case fluctuating between 59 - 61.xxxx, as long as water is present and therefore current flows) and no current flows when there is no water (i.e. 0.0000)? But I just don't know for sure if I'm right...

**Reply by Michsior14** [reply]
- `select.venta_led_strip_mode` - those are the same values that are available on the device control page (http://<device-ip>), but I don't really know their meaning except what is available here: https://github.com/Michsior14/ha-venta/blob/main/resources/settings/106.json#L76-L97
- `sensor.venta_water_level` - I don't think I've ever seen it around 0 (mostly on the lower side of range you mentioned), but I no longer have this model.

**Comment by rgr101** [comment]
Today, I’ve checked the water level sensor when I removed the top for refill : the value went down to ~43.000 and after refill went back to ~60.000. these seem to be the thresholds for ‚dry‘ and ‚wet‘
